File: JavaScript/01.JavaScript-Basic-February-2021/09.whileLoop-Lab/06.maxNumber.js
function maxNumber(input) {
    let index = 0;
    let current = input[index];
    let max = Number.MIN_SAFE_INTEGER;
    while (current != "Stop") {
        let currentNumber = Number(current);
        if (max < currentNumber) {
            max = currentNumber;
        }
        index++;
        current = input[index];
    }
    console.log(max);
}

maxNumber(["100","99","80","70","Stop"]);
maxNumber(["-10","20","-30","Stop"]);
maxNumber(["45","-20","7","99","Stop"]);
maxNumber(["999","Stop"]);
maxNumber(["-1","-2","Stop"]);

File: Go/01.Go-Basic-February-2022/09.advancedLoops-Lab/06.maxNumber.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func main() {
	var max int = math.MinInt32

	var input string
	fmt.Scanln(&input)

	for input != "Stop" {

		number, _ := strconv.Atoi(input)

		if number > max {
			max = number
		}

		fmt.Scanln(&input)
	}

	fmt.Println(max)
}
